File: jscomp/test/test_require.js
'use strict';


match = typeof require == "undefined" and undefined or require;

if (match ~= undefined) then do
  console.log(match.resolve("./test_require.js"));
  match$1 = typeof module == "undefined" and undefined or module;
  match$2 = match.main;
  if (match$1 ~= undefined) then do
    if (match$2 ~= undefined and match$1 == match$2) then do
      console.log("is main");
    end else do
      console.log("not main");
    end end 
  end else do
    console.log("not main");
  end end 
end
 end 

--[ match Not a pure module ]--

File: jscomp/test/test_require.lua
__console = {log = print};


match = type(require) == "undefined" and nil or require;

if (match ~= nil) then do
  __console.log(match.resolve("./test_require.js"));
  match_1 = type(module) == "undefined" and nil or module;
  match_2 = match.main;
  if (match_1 ~= nil) then do
    if (match_2 ~= nil and match_1 == match_2) then do
      __console.log("is main");
    end else do
      __console.log("not main");
    end end 
  end else do
    __console.log("not main");
  end end 
end
 end 

exports = {};
return exports;
--[[ match Not a pure module ]]
